src/windows/export.py

```diff
@@ -374,30 +385,36 @@ def cboSimpleProjectType_index_changed(self, widget, index):
-                        elif "videotoolbox" in codec_text and openshot.FFmpegWriter.IsValidCodec(codec_text):
-                            acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-vtb.svg")
-                        elif "qsv" in codec_text and openshot.FFmpegWriter.IsValidCodec(codec_text):
-                            acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-qsv.svg")
-                        elif openshot.FFmpegWriter.IsValidCodec(codec_text):
-                            acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-none.svg")
+        for preset_folder in [info.EXPORT_PRESETS_PATH, info.USER_PRESETS_PATH]:
+            for file in os.listdir(preset_folder):
+                preset_path = os.path.join(preset_folder, file)
+                try:
+                    xmldoc = xml.parse(preset_path)
+                    type = xmldoc.getElementsByTagName("type")
+
+                    if _(type[0].childNodes[0].data) == selected_project:
+                        titles = xmldoc.getElementsByTagName("title")
+                        videocodecs = xmldoc.getElementsByTagName("videocodec")
+                        for title in titles:
+                            project_types.append(_(title.childNodes[0].data))
+                        for codec in videocodecs:
+                            codec_text = codec.childNodes[0].data
+                            if "vaapi" in codec_text and openshot.FFmpegWriter.IsValidCodec(codec_text):
+                                acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-vaapi.svg")
+                            elif "nvenc" in codec_text and openshot.FFmpegWriter.IsValidCodec(codec_text):
+                                acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-nvenc.svg")
+                            elif "dxva2" in codec_text and openshot.FFmpegWriter.IsValidCodec(codec_text):
+                                acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-dx.svg")
+                            elif "videotoolbox" in codec_text and openshot.FFmpegWriter.IsValidCodec(codec_text):
+                                acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-vtb.svg")
+                            elif "qsv" in codec_text and openshot.FFmpegWriter.IsValidCodec(codec_text):
+                                acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-qsv.svg")
+                            elif openshot.FFmpegWriter.IsValidCodec(codec_text):
+                                acceleration_types[_(title.childNodes[0].data)] = QIcon(":/hw/hw-accel-none.svg")
+
+                except ExpatError as e:
+                    # This indicates an invalid Preset file - display an error and continue
+                    log.error("Failed to parse file '%s' as a preset: %s" % (preset_path, e))
```

Comment:
Part of me wishes this could be some more generic type of exception (like one of the `xml.dom.DOMException` classes), rather than something Expat-specific, since we never _explicitly_ make use of Expat at all — we just call `xml.dom.minidom.parse()` and let it use whatever the available `xml.parsers` implementation is.

But as the docs say, currently there's nothing other than `xml.parsers.expat`, so I suppose it's a moot point.